### PR TITLE
UCP/TAG: Use separate field for md index in offload RNDV hdr

### PR DESCRIPTION
**Pull request by brminich**

### CODE REVIEW

File: src/ucp/tag/offload.c [hunk 1]
```diff
@@ -16,14 +16,13 @@
...
-#define UCP_TAG_OFFLOAD_REQPTR_MDI_MASK  ((uintptr_t)(UCS_SYS_CACHE_LINE_SIZE - 1))
-
 /**
  * Header for unexpected rendezvous
  */
 typedef struct {
     uint64_t       sender_uuid;  /* Sender worker uuid */
-    uintptr_t      reqptr_mdi;   /* Request pointer (high) + md_index (low) */
+    uintptr_t      reqptr;       /* Request pointer */
+    uint8_t        mdi;          /* md index */
```
Comment (yosefe) [inline src/ucp/tag/offload.c]: minor: s/mdi/md_index/

File: src/ucp/tag/offload.c [hunk 2]
```diff
@@ -469,22 +467,24 @@ ucs_status_t ucp_tag_offload_rndv_zcopy(uct_pending_req_t *self)
...
+    ucp_tag_offload_unexp_rndv_hdr_t rndv_hdr = {
+        .sender_uuid   = ep->worker->uuid,
+        .reqptr        = (uintptr_t)req,
+        .mdi           = md_index
     };
 
+    max_rndv = ucp_ep_get_iface_attr(ep, req->send.lane)->cap.tag.rndv.max_hdr;
     dt_state = req->send.state.dt;
 
+    UCS_STATIC_ASSERT(sizeof(ucp_rsc_index_t) <= sizeof(uint8_t));
```
Comment (yosefe) [inline src/ucp/tag/offload.c]: maybe sizeof(rndv_hdr.md_index) instead of sizeof(uint8_t)

File: src/ucp/tag/offload.c [hunk 3]
```diff
@@ -469,22 +467,24 @@ ucs_status_t ucp_tag_offload_rndv_zcopy(uct_pending_req_t *self)
...
-    ucp_request_hdr_t rndv_hdr = {
-        .sender_uuid = ep->worker->uuid,
-        .reqptr      = (uintptr_t)req | md_index
+    ucp_tag_offload_unexp_rndv_hdr_t rndv_hdr = {
+        .sender_uuid   = ep->worker->uuid,
+        .reqptr        = (uintptr_t)req,
+        .mdi           = md_index
     };
 
+    max_rndv = ucp_ep_get_iface_attr(ep, req->send.lane)->cap.tag.rndv.max_hdr;
```
Comment (yosefe) [inline src/ucp/tag/offload.c]: this can be done earlier, in ucp_ep_config_init(), and with ucs_assert_always()

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5888/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5894/ for details (Mellanox internal link).

**Comment by brminich** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5899/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5900/ for details (Mellanox internal link).

**Pull request merged**